disnake/onboarding.py

```diff
@@ -186,11 +211,43 @@ def __repr__(self) -> str:
         )
 
     @property
-    def roles(self) -> List[Role]:
-        """List[:class:`Role`]: A list of roles that will be added to the user when they select this option."""
-        return list(filter(None, map(self.guild.get_role, self.role_ids)))
+    def id(self) -> int:
+        """:class:`int`: The prompt option's ID."""
+        return self._id
 
-    @property
-    def channels(self) -> List[GuildChannel]:
-        """List[:class:`abc.GuildChannel`]: A list of channels that the user will see when they select this option."""
-        return list(filter(None, map(self.guild.get_channel, self.channel_ids)))
+    @classmethod
+    def _from_dict(cls, *, data: OnboardingPromptOptionPayload, guild: Guild) -> Self:
+        if emoji_data := data.get("emoji"):
+            emoji = guild._state.get_reaction_emoji(emoji_data)
+        else:
+            emoji = None
+
+        self = cls(
+            title=data["title"],
+            description=data.get("description"),
+            emoji=emoji,
+            roles=[Object(id=role_id) for role_id in data["role_ids"]],
+            channels=[Object(id=channel_id) for channel_id in data["channel_ids"]],
+        )
+        if "id" in data:
+            self._id = int(data["id"])
+        return self
+
+    def to_dict(self) -> OnboardingPromptOptionPayload:
+        emoji: EmojiPayload = {}  # type: ignore
```

Comment:
this should probably just default to `None` instead.

Reply:
What if we use the default `{id: null, name: null, animated: false}`?

Reply:
Fair point, though that format just seems wrong and I'm still wondering why the API uses it, tbqh.

The client doesn't send an `emoji` field at all when creating options with no emoji, so perhaps we should go with that. I'm hoping this will be made more clear in the api-docs PR, which we have to wait for anyway

Reply:
Fair point, though that format just seems wrong and I'm still wondering why the API uses it, tbqh.

The client doesn't send an `emoji` field at all when creating options with no emoji, so perhaps we should go with that. I'm hoping this will be made more clear in the api-docs PR, which we have to wait for anyway

~~github, why~~

Reply:
now that API docs are here and don't specify which fields have to be present when editing, replicating client behavior would be best as far as I can tell

Reply:
Seems like they now expect a different format here, flattening the emoji fields into the option object: https://github.com/discord/discord-api-docs/pull/6479/files#diff-6a43fc10e10de61f31b90ab22421ef0612d0ea75e3cd53891c6393a6fb84872dR569

Reply:
> Seems like they now expect a different format here, flattening the emoji fields into the option object: [discord/discord-api-docs@bee4b4f](https://github.com/discord/discord-api-docs/commit/bee4b4f52c4afcbb78a3a5842461a33326e5eb2f)

done, 322efc41925298a5c36770f2e5ac8a2b8ec0dcb4